From 8165ff9f315e05307392325fdb8437bf3f5b1f9a Mon Sep 17 00:00:00 2001
From: Matthias Schoepfer <m.schoepfer@rethinkrobotics.com>
Date: Wed, 3 Jan 2024 19:15:11 +0100
Subject: [PATCH] {nanbield} fixes for newer boost

Boost deprecated boost/progress.hpp. This patch fixes the new
location.

Upstream-Status: inappropriate

Signed-off-by: Matthias Schoepfer <m.schoepfer@rethinkrobotics.com>
---
 src/encrypt.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/encrypt.cpp b/src/encrypt.cpp
index da4ece4c29..28b553bba9 100644
--- a/src/encrypt.cpp
+++ b/src/encrypt.cpp
@@ -36,7 +36,7 @@
 
 #include <boost/scoped_ptr.hpp>
 #include <boost/program_options.hpp>
-#include <boost/progress.hpp>
+#include <boost/timer/progress_display.hpp>
 #include <boost/regex.hpp>
 
 #include <ros/ros.h>
@@ -161,9 +161,9 @@ int encrypt(EncryptorOptions const& options)
     outbag.setEncryptorPlugin(options.plugin, options.param);
     outbag.setCompression(options.compression);
     rosbag::View view(inbag);
-    boost::scoped_ptr<boost::progress_display> progress;
+    boost::scoped_ptr<boost::timer::progress_display> progress;
     if (!options.quiet)
-        progress.reset(new boost::progress_display(view.size(), std::cout, "Progress:\n  ", "  ", "  "));
+      progress.reset(new boost::timer::progress_display(view.size(), std::cout, "Progress:\n  ", "  ", "  "));
     for (rosbag::View::const_iterator it = view.begin(); it != view.end(); ++it)
     {
         outbag.write(it->getTopic(), it->getTime(), *it, it->getConnectionHeader());
-- 
2.41.0

